Sources/UIKit/TextAlignmentConstraint.swift

```diff
@@ -0,0 +1,162 @@
+#else
+    import UIKit
+#endif
+
+@objc(BONTextAlignmentConstraintAttribute)
+public enum TextAlignmentConstraintAttribute: Int {
+    case unspecified
+    case top
+    case capHeight
+    case xHeight
+    case firstBaseline
+    case lastBaseline
+    case bottom
+}
+
+private extension TextAlignmentConstraintAttribute {
+    var layoutAttribute: NSLayoutAttribute {
+        switch self {
+        case .top, .capHeight, .firstBaseline, .xHeight:
+            return .top
+        case .lastBaseline, .bottom:
+            return .bottom
+        case .unspecified:
+            return .notAnAttribute
+        }
+    }
+}
+
+private var TextAlignmentConstraintKVOContext = "TextAlignmentConstraintKVOContext" as NSString
+
+@objc(BONTextAlignmentConstraint)
+public class TextAlignmentConstraint: NSLayoutConstraint {
+
+    public private(set) var firstItemAttribute: TextAlignmentConstraintAttribute = .unspecified
+    public private(set) var secondItemAttribute: TextAlignmentConstraintAttribute = .unspecified
+    private var item1: AnyObject!
+    private var item2: AnyObject!
+
+    public static func with(item view1: AnyObject, attribute attr1: TextAlignmentConstraintAttribute, relatedBy relation: NSLayoutRelation, toItem view2: AnyObject, attribute attr2: TextAlignmentConstraintAttribute) -> TextAlignmentConstraint {
+        let constraint = TextAlignmentConstraint(
+            item: view1,
+            attribute: attr1.layoutAttribute,
+            relatedBy: relation,
+            toItem: view2,
+            attribute: attr2.layoutAttribute,
+            multiplier: 1,
+            constant: 0)
+
+        constraint.item1 = view1
+        constraint.item2 = view2
+        constraint.firstItemAttribute = attr1
+        constraint.secondItemAttribute = attr2
+
+        constraint.setupObservers()
+        constraint.updateConstant()
+
+        return constraint
+    }
+
+    deinit {
+        tearDownObservers()
+    }
+
+    public override func awakeFromNib() {
+        super.awakeFromNib()
+        setupObservers()
+    }
+
+    private func setupObservers() {
+        for keyPath in fontKeyPaths {
+            addObserver(self, forKeyPath: keyPath, options: [], context: &TextAlignmentConstraintKVOContext)
+        }
+    }
+
+    private func tearDownObservers() {
+        for keyPath in fontKeyPaths {
+            removeObserver(self, forKeyPath: keyPath, context: &TextAlignmentConstraintKVOContext)
+        }
+    }
+
+    private var fontKeyPaths: [String] {
+        let firstItemFontKeyPath = [
+            #selector(getter: NSLayoutConstraint.firstItem),
+            FontSelector
+            ].map({ $0.description })
+            .joined(separator: ".")
+
+        let secondItemFontKeyPath = [
+            #selector(getter: NSLayoutConstraint.secondItem),
+            FontSelector
+            ].map({ $0.description })
+            .joined(separator: ".")
+
+        return [firstItemFontKeyPath, secondItemFontKeyPath]
```

Comment:
I'm thinking we should also observe `attributedText`, if the object exposes it. If you set the `attributedText` on a `UILabel`, I think it will change `font` without triggering KVO for it.